incentivization.md

```diff
@@ -0,0 +1,206 @@
+
+## Pricing
+
+For MVP, we assume a constant price per hour of history.
+After the client sends a `HistoryQuery` to the server:
+1. The server internally calculates the offer price and sends it to the client.
+2. If the client agrees, it pays and sends a proof of payment to the server.
+3. If the client does not agree, it sends a rejection message to the server.
+4. If the server receives a valid payment before a certain timeout, it sends the response to the client.
+5. If the server receives a rejection message, or receives no message before a timeout, the server assumes that the client has rejected the offer.
+
+### Future work
+
+- DoS protection: a client can overwhelm a server with requests and not proceed to payment. Countermeasure: ignore requests from the same client if they come too often; generalize a reputation system to servers ranking clients.
+- Cost calculation - see https://github.com/waku-org/research/issues/35
+- Price advertisement - see https://github.com/waku-org/research/issues/51
+- Price negotiation - see https://github.com/waku-org/research/issues/52
+
+## Payment
+
+If the client agrees to the price, it sends a _proof of payment_ to the server.
+For the MVP, each request is paid for with a separate blockchain transaction.
+The transaction hash (`txid`) acts as a proof of payment.
+
+Note that client gives proof of payment before it receives the response.
+Other options could be:
+1. the client pays after the fact;
+2. the client pays partly upfront and partly after the fact;
+3. an escrow (a centralized trusted third party, or a semi-trusted entity like a smart contract) ensures atomicity .
+
+Our design considerations are:
+- the MVP protocol should be simple;
+- servers are more "permanent" entities and are more likely to have long-lived identities;
+- it is more important to protect the clients's privacy than the server's privacy.
+
+In light of these criteria, we suggest that the client pays first: this is simpler than splitting the payment, more secure than trusting a third party, and (arguably) more privacy-preserving for the client than the alternative where the client pays after the fact (that would encourage servers to deanonymize clients to prevent fraud).
+
+### Future work
+
+- Add more payment methods - see https://github.com/waku-org/research/issues/58
+- Design a subscription model with service credentials - see https://github.com/waku-org/research/issues/59
+- Add privacy to service credentials - see https://github.com/waku-org/research/issues/60
+
+## Reputation
+
+We use reputation to discourage the server from taking the payment and not responding.
+The client keeps track of the server's reputation:
+- all servers start with zero reputation;
+- if the server honors the request, it gets +1;
+- if the server does not respond before the payment, it gets -1;
+- if the server does not respond after the payment and before a timeout, the client will never query it again.
```

Comment:
I think we're talking about different things here. What I meant was: if Alice pays Bob and then queries him, but he doesn't respond, Bob has essentially ran away with Alice's money, so Alice band him. If there are network issues on Alice's side after she pays, she may wait as long as she wants before querying Bob with that proof of payment. Bob may impose restrictions on the gap between payment and query ("a payment is only valid within 1 year" or something), but for MPV we can assume this gap is infinite.

Reply:
Yeah,Sorry.... i think i did not articulate my comment well enough. 
This was the scenario i was talking about.

A does historyRequestQuery(Q1) to B
B offers a price
A makes payment and sends proof of payment to B
Before B responds with actual data for the query, A looses network connection. 

A reconnects after sometime and wants to request server again for Q1 along with proof-of-payment. I could not see this scenario getting addressed by the simpler protocol laid out and hence was suggesting that we should include the option of indicating proof-of-payment in the historyRequestQuery itself.
 
But now that i think about it, this may lead to complexities. e.g consider another scenario

A does historyRequestQuery (Q1) to B for time t1 to t2(now)
B offers a price
A makes payment and sends proof of payment to B
Before B responds with actual data for the query, A looses network connection. 

Now after sometime when A comes back the historyRequestQuery would be from time t1 to t3(now) where t3>t2.
One solution for this is : B has to indicate to A that it is responding with only data from t1 to t2 and ask for further proof of payment (by specfying the price).

Now i see that this is complicating the price aspect of the protocol.

Reply:
Fair points, thanks! They may become relevant after the MVP phase. I've created a dedicated issue #65.